src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    this.name = name;
+    this.health = health;
+    this.hidden = hidden;
+    Animal.alive.push(this);
+  }
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
```

Comment:
The constructor for Herbivore should not accept 'health' and 'hidden' as parameters, since 'health' defaults to 100 and 'hidden' should be set to false within the constructor.